tests/unit/scribbles/test_transforms.py

```diff
@@ -0,0 +1,310 @@
+        {
+            "scribbles": np.zeros((1, NUM_SLICES, HEIGHT, WIDTH), dtype=np.float32),
+            "roi": [
+                NUM_SLICES // 2 - 4,
+                NUM_SLICES // 2 + 4,
+                HEIGHT // 2 - 8,
+                HEIGHT // 2 + 8,
+                WIDTH // 2 - 16,
+                WIDTH // 2 + 16,
+            ],
+        },
+        (1, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+
+class TestScribblesTransforms(unittest.TestCase):
+    @parameterized.expand(TEST_CASE_ADD_BG_ROI)
+    def test_add_bg_roi_transform(self, input_param, test_input, expected_shape):
+        # float32 test
+        result = AddBackgroundScribblesFromROId(**input_param)(test_input)
+        input_data = test_input["scribbles"].copy()
+        mask = input_data.astype(bool)
+        mask[
+            :,
+            test_input["roi"][0] : test_input["roi"][1],
+            test_input["roi"][2] : test_input["roi"][3],
+            test_input["roi"][4] : test_input["roi"][5],
+        ] = True
+        input_data[~mask] = input_param["scribbles_bg_label"]
+
+        np.testing.assert_equal(input_data, result["scribbles"])
+        self.assertTupleEqual(expected_shape, result["scribbles"].shape)
+        self.assertTupleEqual(test_input["scribbles"].shape, result["scribbles"].shape)
+
+        # int32 test
+        test_input["scribbles"] = test_input["scribbles"].astype(np.int)
+        result = AddBackgroundScribblesFromROId(**input_param)(test_input)
+        input_data = test_input["scribbles"].copy()
+        mask = input_data.astype(bool)
+        mask[
+            :,
+            test_input["roi"][0] : test_input["roi"][1],
+            test_input["roi"][2] : test_input["roi"][3],
+            test_input["roi"][4] : test_input["roi"][5],
+        ] = True
+        input_data[~mask] = input_param["scribbles_bg_label"]
+
+        np.testing.assert_equal(input_data, result["scribbles"])
+        self.assertTupleEqual(expected_shape, result["scribbles"].shape)
+        self.assertTupleEqual(test_input["scribbles"].shape, result["scribbles"].shape)
+
+    @parameterized.expand(TEST_CASE_OPTIM_TX)
+    def test_optimisation_transforms(self, input_param, test_input, output, expected_shape):
+        input_param.update({"post_proc_label": "pred"})
+        for current_tx in [ApplyGraphCutOptimisationd, ApplySimpleCRFOptimisationd]:
+            result = current_tx(**input_param)(test_input)
+            np.testing.assert_allclose(output["target"], result["pred"], rtol=1e-4)
+            self.assertTupleEqual(expected_shape, result["pred"].shape)
+
+    @parameterized.expand(TEST_CASE_ISEG_OPTIM_TX)
+    def test_interactive_graphcut_optimisation_transform(self, input_param, test_input, output, expected_shape):
+        input_param.update({"post_proc_label": "pred"})
+        result = ApplyISegGraphCutPostProcd(**input_param)(test_input)
+        np.testing.assert_allclose(output["target"], result["pred"], rtol=1e-4)
```

Comment:
is there a better alternative to `assert_allclose`?

Reply:
Since these are running optimisers inside, they cannot be guaranteed to have deterministic output. I will check with assert_equal or perhaps we can lower the tolerance.

Reply:
Okay, I have switched these to equal and all seems to work for this simple examples. In future if these pose an issue, we can switch to allclose with a tolerance.